CRM/Member/BAO/Membership.php

```diff
@@ -1515,8 +1515,11 @@ public static function createRelatedMemberships(&$params, &$dao, $reset = FALSE)
       }
       // now go over the queue and create any available related memberships
       reset($queue);
-      while (($available > 0) && ($params = each($queue))) {
-        CRM_Member_BAO_Membership::create($params['value'], $relMemIds);
+      foreach ($queue as $params) {
+        if ($available <= 0) {
+          break;
+        }
+        CRM_Member_BAO_Membership::create($params, $relMemIds);
```

Comment:
when i was testhing this there was no value key in the $params which was causing other test failures

Reply:
Right, so this bit of code hasn't worked for a long time then?
@seamuslee001 It would be nice to rename the `$available` variable to something with a bit more meaning as part of this PR - eg `$numRelatedAvailable`

Reply:
I think you can also drop the `reset($queue);` on line 1517 as it's not required with `foreach`

Reply:
That is my guess but i'm not exactly sure as only looking at it in the prism of unit tests